examples/motoko_examples/calc/canisters/rust/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+use std::cell::RefCell;
+
+thread_local! {
+    static PERF_RESULT_REF_CELL: RefCell<Option<PerfResult>> = RefCell::default();
+}
+
+#[derive(candid::CandidType, Clone)]
+pub struct PerfResult {
+    wasm_body_only: u64,
+    wasm_including_prelude: u64
+}
+
+#[ic_cdk_macros::query]
+pub fn get_perf_result() -> Option<PerfResult> {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| perf_result_ref_cell.borrow().as_ref().cloned())
+}
+
+fn record_performance(start: u64, end: u64) {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| {
+        let mut perf_result = perf_result_ref_cell.borrow_mut();
+
+        *perf_result = Some(PerfResult {
+            wasm_body_only: end - start,
+            wasm_including_prelude: ic_cdk::api::call::performance_counter(0)
+        })
+    });
+}
+
+thread_local! {
+    static CELL: RefCell<i128> = RefCell::new(0);
+}
+
+#[ic_cdk_macros::update]
+pub fn add(n: i128) -> i128 {
+    let perf_start = ic_cdk::api::call::performance_counter(0);
+
+    CELL.with(|counter| *counter.borrow_mut() += n);
+    let result = CELL.with(|counter| counter.borrow().clone());
+
+    let perf_end = ic_cdk::api::call::performance_counter(0);
+
+    record_performance(perf_start, perf_end);
+
+    result
+}
+
+#[ic_cdk_macros::update]
+pub fn sub(n: i128) -> i128 {
+    let perf_start = ic_cdk::api::call::performance_counter(0);
+
+    CELL.with(|counter| *counter.borrow_mut() -= n);
+    let result = CELL.with(|counter| counter.borrow().clone());
+
+    let perf_end = ic_cdk::api::call::performance_counter(0);
+
+    record_performance(perf_start, perf_end);
+
+    result
+}
+
+#[ic_cdk_macros::update]
+pub fn mul(n: i128) -> i128 {
+    let perf_start = ic_cdk::api::call::performance_counter(0);
+
+    CELL.with(|counter| *counter.borrow_mut() *= n);
```

Comment:
Could we possibly combine line 37 and 38? Can you return the counter from the first with?